include/circt/Dialect/Arc/ArcOps.td

```diff
@@ -143,35 +146,40 @@ def StateOp : ArcOp<"state", [
     Optional<I1>:$enable,
     Optional<I1>:$reset,
     I32Attr:$latency,
-    Variadic<AnyType>:$inputs);
+    Variadic<AnyType>:$inputs,
+    Variadic<AnyType>:$initials);
   let results = (outs Variadic<AnyType>:$outputs);
 
   let assemblyFormat = [{
     $arc `(` $inputs `)` (`clock` $clock^)? (`enable` $enable^)?
-    (`reset` $reset^)? `latency` $latency attr-dict
-    `:` functional-type($inputs, results)
+    (`reset` $reset^)?
+    ( `initial` ` ` `(` $initials^ `:` type($initials) `)`)?
+    `latency` $latency attr-dict `:` functional-type($inputs, results)
```

Comment:
Could a `OptionalTypesMatchWith` constraint between the `$initials` types and `$results` types make the `: type($initials)` part redundant? Would be cool if we could get the generated parser to infer the types based on the results :smiley:

Reply:
`OptionalTypesMatchWith` sadly doesn't appear to work with type ranges. `RangedTypesMatchWith` on the other hand refuses to infer `type($initials)` even if I make it non-optional. Maybe it cannot infer an operand range from a result range?!? ODS continues to surprise me for better or worse. I can still imagine that it could be done somehow, but at that point I had thrown in the towel.